compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEDistributeCoresAndObjectFifos.cpp

```diff
@@ -8,32 +8,50 @@
 #include "iree-amd-aie/IR/AMDAIEOps.h"
 #include "iree-amd-aie/Transforms/Passes.h"
 #include "iree-amd-aie/Transforms/Transforms.h"
+#include "iree/compiler/Codegen/TransformStrategies/GPU/Common.h"
 #include "mlir/Dialect/SCF/Transforms/Transforms.h"
 #include "mlir/IR/Iterators.h"
 #include "mlir/Pass/Pass.h"
 #include "mlir/Transforms/GreedyPatternRewriteDriver.h"
 #include "mlir/Transforms/LoopInvariantCodeMotionUtils.h"
 
-#define DEBUG_TYPE "iree-amdaie-unroll-and-distribute-workgroup"
+#define DEBUG_TYPE "iree-amdaie-distribute-cores-and-objectfifos"
 
 namespace mlir::iree_compiler::AMDAIE {
 
+// Used to annotate loops that should be unrolled.
+static const llvm::StringLiteral kAMDAIELoopUnroll = "amdaie.unroll";
+
 namespace {
 
-/// Convert scf.forall ops within a workgroup to scf.for ops
-/// TODO(jornt): use upstream `forallToFor` function once merged.
-LogicalResult workgroupForallToFor(ModuleOp moduleOp) {
+/// Convert local scf.forall ops chosen for parallel distribution to scf.for
```

Comment:
Nit: maybe "inner" scf.forall is a better description?

Reply:
I didn't use `inner` because when we get to include block ids as well, it won't be only the inner ones anymore.